lib/new_relic/agent/opentelemetry/transaction_patch.rb

```diff
@@ -26,7 +21,7 @@ def set_current_segment(new_segment)
             end
 
             # create an otel span for the new segment
-            span = Trace::Span.new(segment: new_segment, transaction: new_segment.transaction)
+            span = Trace::Span.new(span_context: span_context_from_segment(new_segment))
```

Comment:
This is spec compliant with OTel. The only argument in initialize for the span API is `span_context`. Since we're not creating real OTel spans, we just need the context to make sure the OpenTelemetry context calls match our New Relic context calls (ex. current_transaction, current_segment).